pkg/webhooks/workload_webhook.go

```diff
@@ -182,6 +184,87 @@ func validateContainer(c *corev1.Container, path *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateAdmissionChecks(obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i := range obj.Status.AdmissionChecks {
+		admissionChecksPath := basePath.Index(i)
+		ac := &obj.Status.AdmissionChecks[i]
+		allErrs = append(allErrs, validatePodSetUpdates(ac, obj, admissionChecksPath.Child("podSetUpdates"))...)
+	}
+	return allErrs
+}
+
+func validatePodSetUpdates(acs *kueue.AdmissionCheckState, obj *kueue.Workload, basePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	knowPodSets := make(map[string]struct{}, len(obj.Spec.PodSets))
+	knowPodSetNames := make([]string, len(obj.Spec.PodSets))
```

Comment:
No need for this. If you use sets, you can convert them to list with `sets.List`.

Reply:
done https://github.com/kubernetes-sigs/kueue/pull/1220/commits/dd3dde19094c7988fcd3440d59aaf40d1dafee6b